### DISCUSSION TITLE
How to perform an exact search?

### DISCUSSION THREAD

**Discussion by derricck** [winsiderss/systeminformer#1806] [Questions / Answers]
For example:
![image](https://github.com/winsiderss/systeminformer/assets/105561748/58dc20ff-aa9e-423c-b0a3-9fa000f37c0f)

How i could make it match only the "App.exe"?

**Comment by Advait251206** [comment]
As far as I know, the process search box in Process Hacker/System Informer performs a substring match rather than supporting exact-match operators.

In your example, searching for:

```text
app.exe
```

matches both:

```text
App.exe
SearchApp.exe
```

because `App.exe` appears as a substring of `SearchApp.exe`.

I don't believe there is currently a built-in syntax for exact matching, such as:

```text
"App.exe"
```

or

```text
^App\.exe$
```

or

```text
=App.exe
```

If your goal is to find only a single process, a few possible workarounds are:

- Sort by **Name** and manually locate it.
- Search by **PID** once you've identified the process.
- Use **Find Handles or DLLs** if you're searching for something more specific than the process name.
- Add additional columns (e.g., Command Line, Company Name, Path) to help distinguish similarly named processes.

Based on your screenshot, I'd expect the current search behavior to be working as designed: it is finding all processes whose names contain the text `app.exe`, including `SearchApp.exe`.

It would be a useful enhancement to support an exact-match mode (or regex mode) for cases like this, but I don't believe it currently exists.
